### ISSUE TITLE
Make transactions on backend atomic

### ISSUE THREAD

**Issue by epicfaace** [codalab/codalab-worksheets#1504]
Use sqlalchemy's `session` as documented here: https://docs.sqlalchemy.org/en/13/orm/session_transaction.html

**Comment by percyliang** [comment]
Do we have any bugs currently as a result of this?

**Comment by epicfaace** [comment]
No, but I'd imagine it might end up creating some silent bugs / inconsistencies.
